src/useParam.test.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react'
+import useParam from './useParam'
+import { UrlParam } from './params'
+import * as reactRouterDom from 'react-router-dom'
+import { render } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+jest.mock('react-router-dom', () => ({
+  useHistory: jest.fn(() => ({
+    push: jest.fn(),
+  })),
+  useLocation: jest.fn(() => ({
+    search: '',
+  })),
+}))
+
+const countParam: UrlParam<number> = {
+  name: 'count',
+  defaultValue: 2,
+  showDefaultValueInQuery: true,
+  encode: (value) => value.toString(),
+  decode: (value) => parseInt(value, 10),
+}
+
+const complexParam: UrlParam<[[number, number], [number, number]]> = {
+  name: 'complex',
+  defaultValue: [
+    [1, 1],
+    [2, 2],
+  ],
+  encode: (value) => JSON.stringify(value),
+  decode: (value) => JSON.parse(value),
+}
+
+const Component = () => {
```

Comment:
Do we need a component to make this work? Could we not use something like [@testing-library/react-hooks](https://github.com/testing-library/react-hooks-testing-library), like we are in Atlas?